### ISSUE TITLE
add layer normalization

### ISSUE THREAD

**Issue by obo** [ufal/neuralmonkey#105]
Jimmy Lei Ba, Jamie Ryan Kiros, Geoffrey E. Hinton: Layer Normalization
https://arxiv.org/abs/1607.06450
Layer Normalization is a method based on Batch Normalization, but it
works well also for recurrent connections. It substantially speeds up
training and allows reaching slightly better performance (BatchNorm also
acts as a regularizer able to replace dropout, but this effect of
LayerNorm is not discussed in the paper).

**Comment by loveJasmine** [comment]
any example to enable/disable this feature in ini config file for NMT/chatbot task?

**Comment by tomkocmi** [comment]
As to my knowledge, it isn't implemented in the NM yet. You can implement it by yourself, then we will be glad if you create pull request to include it in the NM

**Comment by jindrahelcl** [comment]
There is a function in TF that does layer normalization. See transformer encoder & decoder. Incorporating of layer normalization to the recurrent scheme will be done as part of #648
